packages/dapp/src/components/ConnectionModal/CreateConnectionModal.tsx

```diff
@@ -17,19 +24,58 @@ const CreateConnectionModal = ({
   setOpen,
 }: CreateConnectionModalProps) => {
   const t = useTranslations('CreateConnectionModal');
+
+  // Local state
   const [connectionData, setConnectionData] = useState<string | null>(null);
-  const { request, session, changeRequest, changeSession } = useSessionStore(
-    (state) => ({
-      request: state.request,
-      session: state.session,
-      changeRequest: state.changeRequest,
-      changeSession: state.changeSession,
-    })
-  );
 
-  const createSession = async (): Promise<string> => {
+  // Global state
+  const { token, isSignedIn } = useAuthStore((state) => ({
+    token: state.token,
+    isSignedIn: state.isSignedIn,
+  }));
+
+  const {
+    changeSession,
+    changeSessionId,
+    changeConnected,
+    changeHasCamera,
+    changeDeviceType,
+  } = useEncryptedSessionStore((state) => ({
+    changeSession: state.changeSession,
+    changeSessionId: state.changeSessionId,
+    changeConnected: state.changeConnected,
+    changeHasCamera: state.changeHasCamera,
+    changeDeviceType: state.changeDeviceType,
+  }));
+
+  const createSession = async (): Promise<string | null> => {
+    if (!token || !isSignedIn) {
+      return null;
+    }
+
     // Create session ID
-    const sessionId = crypto.randomUUID();
+    const createSessionResult = await fetch('/api/encrypted-session', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    if (!createSessionResult.ok) {
+      setTimeout(() => {
+        useToastStore.setState({
+          open: true,
+          title: 'Failsed to create session. Please try again.',
```

Comment:
Typo here.

Reply:
Also maybe localize this, since we're planning to reformat all toasts to consistent localized versions.